additional docker configs for local development

gsalsero opened the pull request:
Added the following env vars to help with local development

- FTP_USERNAME
- FTP_PASSWORD
- FTP_PASSIVE_MODE
- FTP_OVERWRITE
- GAMES_LIMIT
- EXCLUDED_PLAYERS

provided defaults for the following to help with local development 
- MYSQL_DATABASE
- MYSQL_USER
- MYSQL_PASSWORD
- DB_USERNAME
- DB_PASSWORD

gsalsero commented:
considering the latest changes made on master, I need to see if these changes still make sense. I will make this as a draft in the meantime.

gsalsero commented:
> considering the latest changes made on master, I need to see if these changes still make sense. I will make this as a draft in the meantime.

only minor changes were needed

evilru commented on docker-compose.local.yml:
I would not add default values here, but comment out unused variables,
then the defaults from the dockerfile take over.
but then I am asking myself why should they be listet und commented out?

gsalsero replied:
I'll remove the defaults.

evilru commented on vutil.php:
no php file should have a closing a tag, this might lead to unwanted whitespace.
another situation would be when it outputs html as well, but this does not.

evilru commented:
I took your input and restructured the current setup so you can add your changes easily.
You might want to base your change on this commit: d3676c8fec73f0a8537f61e3b5025f3c85104688

I dropped the override file as it is not needed and moved the dev values to the local compose where it reads values from a not commited .env file.

this was great input to bring a better structure to the project.
I did not touch any of the additional variables and changes you thought of.

beside, with the last commits, I introduced a dev container including xdebug.
this was the reason I dropped it from the compose.

the dev container misses docker for now, I will add it later

evilru reviewed (changes requested):
please have a look at the comments I added to the files.
I hope everything is clear to understand.

gsalsero commented:
> I took your input and restructured the current setup so you can add your changes easily. You might want to base your change on this commit: [d3676c8](https://github.com/evilru/quake3-vsp-stats/commit/d3676c8fec73f0a8537f61e3b5025f3c85104688)
> 
> I dropped the override file as it is not needed and moved the dev values to the local compose where it reads values from a not commited .env file.
> 
> this was great input to bring a better structure to the project. I did not touch any of the additional variables and changes you thought of.
> 
> beside, with the last commits, I introduced a dev container including xdebug. this was the reason I dropped it from the compose.
> 
> the dev container misses docker for now, I will add it later

I noticed that the `.env` file was added to the repository. Typically, this file is excluded from version control because it may contains sensitive environment-specific details. Instead, a sample file like `.env.sample` is usually provided. I use my local .env file to adjust environment settings without having those changes tracked, which helps avoid accidentally committing sensitive data—such as FTP user credentials. Could we consider removing the .env file from the repo and using a sample file instead?

gsalsero commented:
> please have a look at the comments I added to the files. I hope everything is clear to understand.

Thanks, I'll familiarize myself with the other change and perhaps close this PR if appropriate.

gsalsero commented on docker-compose.yml:
That makes sense.

gsalsero commented on Dockerfile:
```diff
@@ -67,7 +72,8 @@ RUN chmod +x docker/import.sh \
  && sed -ri -e 's!(\['\''default_skin'\''\]\s*=\s*)'\''fest'\''(;)!\1$_ENV["DEFAULT_SKIN"]\2!g' pub/themes/bismarck/settings.php \
- && sed -ri -e 's!(\['\''check_unique_gameID'\''\]\s*=\s*)1!\1$_ENV["CHECK_UNIQUE_GAMEID"]!g' pub/configs/cfg-default.php
+ && sed -ri -e 's!(\['\''check_unique_gameID'\''\]\s*=\s*)1!\1$_ENV["CHECK_UNIQUE_GAMEID"]!g' pub/configs/cfg-default.php \
+ && sed -ri -e '/\$player_exclude_list=array\(/,/\);/c\$player_exclude_list=explode\('\'','\'', $_ENV["EXCLUDED_PLAYERS"]\);' pub/include/playerExcludeList-default.inc.php
```
I can see that. I would still want the ability to update the values without it showing up as a change in git.

gsalsero commented on Dockerfile:
I'll do that.

evilru commented:
> > I took your input and restructured the current setup so you can add your changes easily. You might want to base your change on this commit: [d3676c8](https://github.com/evilru/quake3-vsp-stats/commit/d3676c8fec73f0a8537f61e3b5025f3c85104688)
> > I dropped the override file as it is not needed and moved the dev values to the local compose where it reads values from a not commited .env file.
> > this was great input to bring a better structure to the project. I did not touch any of the additional variables and changes you thought of.
> > beside, with the last commits, I introduced a dev container including xdebug. this was the reason I dropped it from the compose.
> > the dev container misses docker for now, I will add it later
> 
> I noticed that the `.env` file was added to the repository. Typically, this file is excluded from version control because it may contains sensitive environment-specific details. Instead, a sample file like `.env.sample` is usually provided. I use my local .env file to adjust environment settings without having those changes tracked, which helps avoid accidentally committing sensitive data—such as FTP user credentials. Could we consider removing the .env file from the repo and using a sample file instead?

Thank you - I just removed it and added it to .gitignore
